integration/exec.go

```diff
@@ -0,0 +1,69 @@
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// RunBinary runs a target binary with a set of arguments provided by the user
+func RunBinary(binPath string, args ...string) (func(), error) {
+	abs, err := filepath.Abs(binPath)
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	cmd := exec.CommandContext(ctx, abs, args...)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	if err := cmd.Start(); err != nil {
+		return nil, err
+	}
+	return cancel, nil
+}
+
+// BuildSource builds a target Go package and gives the resulting binary
+// some user-defined name
+func BuildGoSource(packagePath, output string) (func() error, error) {
+	cmdBuild := exec.Command("go", "build", "-o", output, packagePath)
+	if out, err := cmdBuild.CombinedOutput(); err != nil {
+		return nil, errors.New(fmt.Sprintf("unable to build package: %v (%s)",
+			err, string(out),
+		))
+	}
+	return func() error {
+		return os.Remove(output)
+	}, nil
+}
+
+// RunContainer launches a detached docker container on the host machine. It
+// takes an image name, a list of "docker run" arguments, and a list of
+// arguments that get passed to the container runtime
+func RunContainer(image string, dockerArgs, runtimeArgs []string) (func() error, error) {
```

Comment:
Nice idea! I'll take a look. I would definitely prefer using the Docker API because system commands feel clunky for some reason. Hopefully it'll make testing much easier, too.